src/Client/components/profile/ProfileEdit.tsx

```
@@ -1,105 +1,168 @@
+  });
+  const { setProfileImage, profileImage } = useAuth();
+  const [imageUrl, setImageUrl] = useState<string | null>(null);
+  const [uploading, setUploading] = useState(false);
+  const isControlled = formData && setFormData;
+  const currentFormData = isControlled ? formData : localFormData;
   const updateFormData = useCallback(
-    (updateFn: (prev: FormData) => FormData) => {
+    (updateFn: (prev: ProfileFormData) => ProfileFormData) => {
       if (isControlled) {
-        setFormData!(updateFn)
+        setFormData!(updateFn);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)